compiler/grainformat/reformat.re

```diff
@@ -466,147 +203,861 @@ let prefixop = (op: string) => {
   };
 };
 
-let rec resugar_list_patterns =
-        (
-          ~patterns: list(Parsetree.pattern),
-          ~parent_loc,
-          ~original_source: array(string),
-        ) => {
-  let processed_list = resugar_pattern_list_inner(patterns, parent_loc);
-  let last_item_was_spread = ref(false);
+let no_attribute = _ => Doc.nil;
 
-  let items =
-    List.map(
-      i =>
-        switch (i) {
-        | RegularPattern(e) =>
-          last_item_was_spread := false;
+let remove_used_comments =
+    (
+      ~all_comments: list(Parsetree.comment),
+      remove_comments: list(Parsetree.comment),
```

Comment:
This will usually be a small list we are manipulating, but I take your point.  It's almost always removing the comment we wrote after the opening brace from the remaining comments which I could optimise on

Reply:
Labelled comments fixed.